pkg/security/ImageScanService.go

```diff
@@ -716,33 +759,57 @@ func (impl *ImageScanServiceImpl) HandleProgressingScans() {
 	}
 
 	var executionHistoryDirPath string
+	flagForDeleting := false
 	// Create Folder for output data for execution history only if any pending scans are there due to pod died
 	if len(scanHistories) > 0 {
+		flagForDeleting = true
 		executionHistoryDirPath = impl.CreateFolderForOutputData(scanHistories[0].ImageScanExecutionHistoryId)
 	}
 	wg := &sync.WaitGroup{}
 	wg.Add(len(scanHistories))
-	// System doing image scanning for all pending scans
+	imagescanExecutionHistories, err := impl.scanHistoryRepository.FindAll()
+	if err != nil {
+		impl.logger.Errorw("error in getting scan histories on start up", "err", err)
+		return
+	}
+	imageScanToolMetadatas, err := impl.scanToolMetadataRepository.FindAllActiveTools()
+	if err != nil {
+		impl.logger.Errorw("error in getting all active tools", "err", err)
+	}
+	imageScanExecutionHistoryMap := make(map[int]*repository.ImageScanExecutionHistory)
+	imageScanToolsMap := make(map[int]*repository.ScanToolMetadata)
+
+	for _, imageScanExecutionHistory := range imagescanExecutionHistories {
+		imageScanExecutionHistoryMap[imageScanExecutionHistory.Id] = imageScanExecutionHistory
+	}
+	for _, imageScanToolMetaData := range imageScanToolMetadatas {
+		imageScanToolsMap[imageScanToolMetaData.Id] = imageScanToolMetaData
+	}
+
+	//System doing image scanning for all pending scans
 	for _, scanHistory := range scanHistories {
 		scanEvent := common.ImageScanEvent{}
-		err = json.Unmarshal([]byte(scanHistory.ImageScanExecutionHistory.ScanEventJson), &scanHistory)
+		err = json.Unmarshal([]byte(imageScanExecutionHistoryMap[scanHistory.ImageScanExecutionHistoryId].ScanEventJson), &scanHistory)
```

Comment:
don't directly use map, get string separately with nil pointer check and then use it